docs: Add and refine developer contribution guidelines (#1356)
omegat-org/omegat: src_docs/developer/00.ContributingEcoSystem.md

---

## Documentation

Anyone can contribute to OmegaT's documentation, whether by writing a user manual, creating tutorial videos, or improving the existing help files. If the material is useful, the OmegaT team will gladly share it with the broader user community.

We value contributions of all types — not just code.

---

## Localization

Localization (L10N) of the application and documentation is an active area of contribution. If you'd like to help translate OmegaT into your language, we invite you to join the OmegaT-L10N project on GitHub:  
👉 [https://github.com/OmegaT-L10N](https://github.com/OmegaT-L10N)

Details about how to get involved are provided in the Localization section of this guide.

---

## Adoption of Contributions

Everyone is welcome to suggest changes to OmegaT. The project team reviews these suggestions and includes those that benefit most users in the official releases. This helps maintain the quality and reliability of OmegaT.

Even if your suggestion isn't included in the official release, remember that you can still modify your own copy of OmegaT or share your modified version with others (following the GPL-3 license terms).

> This flexibility is one of the great benefits of open-source software.

Author: ```suggestion
This flexibility is one of the great benefits of free software.
```

# Variety of Contribution

## The OmegaT Project Philosophy

OmegaT is a free, libre, and open-source software (FLOSS) project. This means that:
- You can download and use it at no cost
- You can see and modify the source code
- You can share the program with others

OmegaT uses the GNU General Public License version 3 (GPL-3), which gives you the freedom to:
- Use the program for any purpose
- Study and change the program to suit your needs
- Share the program with friends and colleagues
- Share your improvements with others

These freedoms are central to the OmegaT philosophy.

> It is often referred to as "free as in speech" as opposed to "free as in beer"

Author: ```suggestion
```

---

## Documentation

Anyone can contribute to OmegaT's documentation, whether by writing a user manual, creating tutorial videos, or improving the existing help files. If the material is useful, the OmegaT team will gladly share it with the broader user community.

We value contributions of all types — not just code.

---

## Localization

Localization (L10N) of the application and documentation is an active area of contribution. If you'd like to help translate OmegaT into your language, we invite you to join the OmegaT-L10N project on GitHub:  
👉 [https://github.com/OmegaT-L10N](https://github.com/OmegaT-L10N)

Details about how to get involved are provided in the Localization section of this guide.

---

## Adoption of Contributions

Everyone is welcome to suggest changes to OmegaT. The project team reviews these suggestions and includes those that benefit most users in the official releases. This helps maintain the quality and reliability of OmegaT.

Even if your suggestion isn't included in the official release, remember that you can still modify your own copy of OmegaT or share your modified version with others (following the GPL-3 license terms).

> This flexibility is one of the great benefits of open-source software.

Reviewer: Yes. References to "open source" are superfluous. We should refer to OmegaT as "free software" and compatible software as "free software" too. We need to focus on explaining what "freedom" means.

I’ll make a few proposals later. Thank you for working on this.

---

## Communication Channels

OmegaT is developed by volunteers who contribute based on their interests and skills.
Everyone works on what they find most interesting or important.

Here's how you can connect with the community:

- **User Support Mailing List**  

>   A friendly place where both new and experienced users can ask questions and suggest improvements.

Author: ```suggestion
  A friendly place where both new and experienced users can ask questions and suggest improvements.  
  → [Search/Subscribe the list](https://sourceforge.net/projects/omegat/lists/omegat-users)
```

- Don't be shy - this is the main place for getting help and sharing ideas.

- **Developers Mailing List**  

>   For discussions about code and technical aspects of OmegaT.

Author: ```suggestion
  For discussions about code and technical aspects of OmegaT.
    → [Search/Subscribe the list](https://sourceforge.net/projects/omegat/lists/omegat-development)
```

- If you're interested in the programming side of things, this is the place to join.

- **Issue Tracker**  

>   Where you can report bugs or suggest new features (called RFEs: Requests for Feature Enhancement).

Author: ```suggestion
  Where you can report bugs or suggest new features (called RFEs: Requests for Feature Enhancement).
    → [Search/Record the issue](https://sourceforge.net/p/omegat/_list/tickets)
```

# Variety of Contribution

## The OmegaT Project Philosophy

OmegaT is a free, libre, and open-source software (FLOSS) project. This means that:
- You can download and use it at no cost
- You can see and modify the source code
- You can share the program with others

OmegaT uses the GNU General Public License version 3 (GPL-3), which gives you the freedom to:
- Use the program for any purpose
- Study and change the program to suit your needs
- Share the program with friends and colleagues

> - Share your improvements with others

Author: The latter four bullets are the four principles of the Free Software.
Could you propose a merged change?

These freedoms are central to the OmegaT philosophy.

> It is often referred to as "free as in speech" as opposed to "free as in beer"

Author: Is it ok to remove "It is often referred to as "free as in speech" as opposed to "free as in beer"" ?

> It is often referred to as "free as in speech" as opposed to "free as in beer"

Reviewer: Indeed. The "beer" item is an English cultural reference and we need to focus on freedom here.
